lib/common/http2client.c

```diff
@@ -946,31 +946,46 @@ static void on_connection_ready(struct st_h2o_http2client_stream_t *stream, stru
     h2o_url_t url;
     h2o_header_t *headers;
     size_t num_headers;
-    h2o_iovec_t body;
+    h2o_httpclient_req_body_t body = {H2O_HTTPCLIENT_REQ_BODY_NONE};
     h2o_httpclient_properties_t props = (h2o_httpclient_properties_t){NULL};
 
     register_stream(stream, conn);
 
     stream->super._cb.on_head =
         stream->super._cb.on_connect(&stream->super, NULL, &method, &url, (const h2o_header_t **)&headers, &num_headers, &body,
-                                     &stream->streaming.proceed_req, &props, &conn->super.origin_url);
+                                     &props, &conn->super.origin_url);
     if (stream->super._cb.on_head == NULL) {
         close_stream(stream);
         return;
     }
 
+    {
+        h2o_headers_t headers_vec = (h2o_headers_t){headers, num_headers, num_headers};
+        h2o_httpclient__add_cl_or_te_header(stream->super.pool, method, &headers_vec, &body, NULL);
+        headers = headers_vec.entries;
+        num_headers = headers_vec.size;
+    }
```

Comment:
Same as the case for http1client. I'd prefer passing the value of the content-length header (should it be sent) to `h2o_hpack_flatten_request`. There's actually a benefit in doing that, as we could use the [Literal Header Field without Indexing](https://httpwg.org/specs/rfc7541.html#literal.header.without.indexing) encoding for the Content-Length header field. The rationale is that there is no point retaining the Content-Length header fields in the dynamic table.